### ISSUE TITLE
Bulk change on a new many 2 many related record creates several records in the related table intead of one

### ISSUE THREAD

**Issue by asitemade4u** [directus/app#2685]
# Bug Report

## Steps to Reproduce

1. We have 2 Collections, A and B. They are related many to many using an automatically defined (by Directus) intermediary table
1. in the A table, while creating the m2m relationship, I have allowed "Create" in the linked table (= B)
1. I select several records in A and click the bulk change button (pencil)
1. I create a new record in B using the **Add New** button (seems to work fine, as the record in table B seems to be created indeed)
1. I submit the change in bulk using the **Submit** button

## Expected Behavior
There should be one record created in B, linked to several records in A using the intermediary table.

## Actual Behavior
Instead there are as many records created in B as I have records in A modified in bulk. That is not a m2m relationship but a 1 to 1 instead.
NB. It works fine when the related record already exists and is picked using the **Select Existing** button

## Technical Details

- Device: Desktop, Firefox browser
- OS: Debian 10
- Web Server: [eg: Apache 2.4.37]
- Last version of Directus from the Docker install (kept updated using Watchtower). Local server on Debian 10

**Comment by rijkvanzanten** [comment]
This is a very interesting one, I'm not sure what I expected to happen in this case 🤔 

> There should be one record created in B, linked to several records in A using the intermediary table.

@benhaynes does that sound right to you? If you batch edit collection "articles" and you add a new (m2m) category "nature", you'd expect only one category to have been created and linked to all these articles, but likewise, if you're working with orders and invoices, you don't want to have the same invoice for every order 🤔

**Comment by asitemade4u** [comment]
Yes, agreed, as would have happened if I had created the record in B 
then added it from the bulk change GUI.

**Comment by rijkvanzanten** [comment]
Selecting an existing one in bulk makes total sense to me, creating a new one can have two different desired effects I think..

**Comment by asitemade4u** [comment]
Not sure about that, at least not what common sense expects : we are in 
a many to many relationship.

**Comment by benhaynes** [comment]
This is certainly an interesting one. It does make more sense (maybe 80/20?) to link the single added item to all the batch edited items. I'm trying to think of more examples of the other way (individual records duplicated) especially since those new records would all be the exact same.

So my ordered recommendations would be:
1. Save a single related item and then link that same record to all relations
2. Offer two options after adding the item: `Add Single Item to All` or `Duplicate Item for Each`
3. Disable adding new for batch editing (could be a stop-gap solution, but not long-term. If we do this we need to add a Feature Request to circle back later.

**Comment by asitemade4u** [comment]
Yes, the procedure you suggest works fine and I have adopted it since I 
discovered the current behavior.

If I may add my two cents, what Rijk says is not in line with the reason 
why we have relational databases in the first place: to avoid record 
redundancy. If you create several linked records with the exact same 
content, you are deep down into record redundancy ;o))

**Comment by rijkvanzanten** [comment]
Yeah that's very true, just making sure by replacing the current way it works we upset other's who expect / rely on it to work that way 🤷‍♂

**Comment by benhaynes** [comment]
Yes and no — I agree that the main purpose of a relational database is to normalize and avoid duplication (DRY). BUT, there are workflows that might be outside of the process you're describing... and they need to be considered.

For instance, a user might be using the bulk feature to _start drafts_ of similar items that need to be _individually_ linked. So they might go through this process to add the boilerplate data... then open the individually created items to finish/tailor each. I'm not saying this is common, or is important enough to build-around... just that we really have to avoid snap judgements in a CMS that aims to be as unopinionated as possible.

In any event, I can't think of many examples of wanting to create these "starter/draft" items... so I think we should just go with my option #1.

**Comment by asitemade4u** [comment]
In that case, one would not use many to many relationships but many to one.

**Comment by benhaynes** [comment]
No, you're misunderstanding what I'm saying... it's a M2M, and there are multiple items linked. They would be using this bulk feature to create a set of new items with _initial data_ (that is the same across all), and then they could edit those items to make them unique. 

Imagine you have 10 articles you're bulk editing, and you add a new "gallery" (multiple images). What I'm saying is that there's a chance they want to create individual galleries that are all have the same first 5 images... and then after saving all that, they would go into each individual relational gallery and add 10 more images to make them unique.

This is a silly example, but I don't have time to think of a better one right now. The point is, we need to think of ALL use-cases (even edge-cases) of our platform before making far-reaching decisions. Otherwise we'll get another feature request in a few months asking to add the opposite.

**Comment by rijkvanzanten** [comment]
> Otherwise we'll get another feature request in a few months asking to add the opposite.

This is exactly what I was getting at above as well ^.^
